### DISCUSSION TITLE
fragments left after replacing a dead node with a new device

### DISCUSSION THREAD

**Discussion by Thaui57** [zwave-js/zwave-js-ui#2486] [Q&A]
Like to know how I can remove some fragments from a replaced dead node which I replaced with a new node. After each restart of my homesystem (Domoticz) the old device names will be discovered beside the new existing device. I have not found any possibility to find out where to delete the data. In Domoticz I have deleted the fragments and the system seems to be clean. But if I reboot the old data appears again. I have 6 device with this behaviour. The replaced device under the old node ID is working correct.

!--StartFragment--

  | 381 | Razberry7pro | zwavejs2mqtt_0xdbf01f81_4-51-0-currentColor | 1 | Deckenlampe-Licht AZ LED_rgb_dimmer | Color Switch | RGBWZ | Off | - | - |   | 2022-06-26 14:55:28
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
  |   | 382 | Razberry7pro | zwavejs2mqtt_0xdbf01f81_4-38-1-currentValue | 1 | Deckenlampe-Licht AZ LED_dimmer_1 | Light/Switch | Switch | Off | - | - |   | 2022-06-26 14:55:28
  |   | 383 | Razberry7pro | zwavejs2mqtt_0xdbf01f81_4-38-2-currentValue | 1 | Deckenlampe-Licht AZ LED_dimmer_2 | Light/Switch | Switch | Off | - | - |   | 2022-06-26 14:55:28
  |   | 384 | Razberry7pro | zwavejs2mqtt_0xdbf01f81_4-38-3-currentValue | 1 | Deckenlampe-Licht AZ LED_dimmer_3 | Light/Switch | Switch | Off | - | - |   | 2022-06-26 14:55:28
  |   | 385 | Razberry7pro | zwavejs2mqtt_0xdbf01f81_4-38-4-currentValue | 1 | Deckenlampe-Licht AZ LED_dimmer_4 | Light/Switch | Switch | Off | - | - |   | 2022-06-26 14:55:28
  |   | 386 | Razberry7pro | zwavejs2mqtt_0xdbf01f81_4-38-5-currentValue

!--EndFragment--

**Comment by robertsLando** [comment]
What do you mean with 'fragments'?

**Reply by Thaui57** [reply]
Node number 4 is no longer existing but this 6 devices are still appering after each new reboot. So I guess this are fragments in the store and will be re-discovered when starting the system because I delete them in the Domoticz data base.

**Reply by robertsLando** [reply]
Try removing failed nodes, be aware that if you have battery devices this could remove them too

**Reply by Thaui57** [reply]
I guess I have to start from the beginning with my problem.
I had a node no.4 which failed. Instead to remove the node I have replaced node no.4 with another device. After including the new device i mentioned that some fragments from the old device are still present. To solve the problem I excluded the newly installed device again hoping that everything from node no. 4 will be erased. But that's not the case. The above mentioned idx are still there i guess. Because after each reboot they appear again. How can I get rid of this fragmemts?

**Reply by robertsLando** [reply]
Just try to call the api from advanced dialog named  'remove failed nodes'

**Reply by AlCalzone** [reply]
![image](https://user-images.githubusercontent.com/17641229/176610233-d8607f2a-1f13-4f57-862e-75b2521a1e01.png)

**Reply by Thaui57** [reply]
thanks for the remarks. I did that already but repeated the action due to your advise. Here is the Error message. 
Error while calling API removeFailedNode. the node removal process could not be started because the node responded to a ping(ZW0360).
I am on ZwaveJS2Mqtt version 6,13,0  ZwaveJS 9.5.0
No change after deleting the  zwavejs2mqtt_0xdbf01f81_4- in Domotice and  restarting the system (reboot). The device will appear again

**Reply by AlCalzone** [reply]
> because the node responded to a ping

Then the node is still existing and alive and I don't understand your problem. Maybe explain it with screenshots, describing what is expected and what actually is.

**Reply by Thaui57** [reply]
here are some pictures to understand better what I mean. I do not have a node No.4. The discovery list from Domoticz is showing ID's with a node no.4. The real node is node no.19. The question is how do I get rid of the fragments. 
![image](https://user-images.githubusercontent.com/90320300/176840670-3fe07b19-1b08-49f6-b360-f966dc29c9fa.png)
![image](https://user-images.githubusercontent.com/90320300/176841084-82a73f0a-fbeb-43c1-888a-fac1faabcdbf.png)

**Reply by AlCalzone** [reply]
Okay, I'll need to see a driver log from the startup of zwavejs2mqtt, on loglevel debug. Please attach it as a file here.
[Here's a description](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs) how to do that.

During your attempt at removing the failed node, there was definitely a response from a node, so **something** still has ID 4.

**Reply by Thaui57** [reply]
when I changed the General like in the description I get this page. I do not have a PW for zwavejs2Mqtt
![image](https://user-images.githubusercontent.com/90320300/176917697-27b51505-1002-4592-aef7-d66d0aec8964.png)

**Reply by AlCalzone** [reply]
@robertsLando ?

**Reply by robertsLando** [reply]
I think he has enabled auth in general section by error. BTW the default user and psw are `admin` and `zwave`

**Comment by Thaui57** [comment]
as requested attached find the log file after booting and the dump file of all nodes

 

Von: AlCalzone <> 
Gesendet: Freitag, 1. Juli 2022 10:36
An: zwave-js/zwavejs2mqtt ***@***.***>
Cc: Thaui57 ***@***.***>; Author ***@***.***>
Betreff: Re: [zwave-js/zwavejs2mqtt] fragments left after replacing a dead node with a new device (Discussion #2486)

 

Okay, I'll need to see a driver log from the startup of zwavejs2mqtt, on loglevel debug. Please attach it as a file here.
Here's a description <https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs>  how to do that.

During your attempt at removing the failed node, there was definitely a response from a node, so something still has ID 4.

**Reply by AlCalzone** [reply]
You'll need to upload the file via the Github UI. Attaching it to the email doesn't work.

**Reply by Thaui57** [reply]
[zwavejs_2022-07-02.log](https://github.com/zwave-js/zwavejs2mqtt/files/9039955/zwavejs_2022-07-02.log)

**Reply by Thaui57** [reply]
[nodes_dump.log](https://github.com/zwave-js/zwavejs2mqtt/files/9039968/nodes_dump.log)

**Reply by AlCalzone** [reply]
There are no references at all to a node 4 in the log or the nodes dump. I'm not sure how Domoticz determines which data to show.
Can you verify if zwavejs2mqtt shows anything regarding Node 4? If not, I think this is a bug in Domoticz.

**Reply by robertsLando** [reply]
@Thaui57 I think you may have some nodes left there from retained topics, try to clean up them using mqtt explorer,

**Comment by Thaui57** [comment]
Thx. Will doAm 07.07.2022 08:13 schrieb Daniel Lando ***@***.***>:
@Thaui57 I think you may have some nodes left there from retained topics, try to clean up them using mqtt explorer,
